### PR TITLE
3716 enhance doc-strings of DenseNet

### PR DESCRIPTION
**Pull request by yiheng-wang-nv**
Signed-off-by: Yiheng Wang <vennw@nvidia.com>

Related to #3716  .

### Description
In this PR, I enhanced the doc-strings of DenseNet to explain its non-determinism.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/networks/nets/densenet.py [hunk 1]
```diff
@@ -148,6 +148,9 @@ class DenseNet(nn.Module):
     """
     Densenet based on: `Densely Connected Convolutional Networks <https://arxiv.org/pdf/1608.06993.pdf>`_.
     Adapted from PyTorch Hub 2D version: https://pytorch.org/vision/stable/models.html#id16.
+    This network is non-determistic When `spatial_dims` is 3 and CUDA is enabled. Please check the link below
```
Comment (wyli) [inline monai/networks/nets/densenet.py:151]: I think it'll only affect the training (with backprop), the forward part is deterministic could you help double check? Thanks!

**Review by wyli — APPROVED** [review]

File: monai/networks/nets/densenet.py (hunk 1 above)
Comment (yiheng-wang-nv) [reply monai/networks/nets/densenet.py:151]: Thanks @wyli , yes, if only do forward part, it is not affected. Should it be mentioned in the doc-strings?

Comment (wyli) [reply monai/networks/nets/densenet.py:151]: thank you, I think the docstring is fine...

**Pull request merged**